modules/internal/ChapelIO.chpl

```diff
@@ -257,7 +257,7 @@ module ChapelIO {
       var isBinary = writer.binary();
 
       if (isClassType(t)) {
-        if t != object {
+        if t != borrowed object {
```

Comment:
Please double-check that all class types arriving here are "borrowed".

Seems like they are, because this is invoked from writeThis() that's generated by compiler's buildDefaultReadWriteFunctions(AggregateType* ct), and an AggregateType means it's a "borrow" ?

What if the user calls writeThisFieldsDefaultImpl()?

Maybe worth adding compilerAssert(t is a borrowed type).
Or just convert `t` to `borrowed` before comparing against `borrowed object`.